RCAIDE/Methods/Propulsion/propeller_design.py

```diff
@@ -0,0 +1,303 @@
+from Legacy.trunk.S.Core.Utilities import interp2d
+import numpy as np
+import scipy as sp 
+from scipy.optimize import root 
+
+# ----------------------------------------------------------------------
+#  Propeller Design
+# ----------------------------------------------------------------------
+
+def propeller_design(prop, tol=1e-10):
+    """ Optimizes propeller chord and twist given input parameters.
+          
+          Inputs:
+          
+          prop.
+            hub radius                       [m]
+            tip radius                       [m]
+            rotation rate                    [rad/s]
+            freestream velocity              [m/s]
+            number of blades               
+            number of stations
+            design lift coefficient
+            design_altitude                  [m]
+            airfoil data
+          airfoil_analysis                    Analysis method for airfoils
+          number_of_stations                 Number of blade element stations
+          tol                                Convergence tolerance for rotor design
+            
+          Outputs:
+          Twist distribution                 [array of radians]
+          Chord distribution                 [array of meters]
+              
+          Assumptions/ Source:
+          Based on Design of Optimum Propellers by Adkins and Liebeck
+          
+    """
+    # Unpack
+    N            = prop.number_of_radial_stations       # this number determines the discretization of the propeller into stations
+    B            = prop.number_of_blades
+    R            = prop.tip_radius
+    Rh           = prop.hub_radius
+    omega        = prop.angular_velocity    # Rotation Rate in rad/s
+    V            = prop.freestream_velocity # Freestream Velocity
+    Cl           = prop.design_Cl           # Design Lift Coefficient
+    alt          = prop.design_altitude
+    Thrust       = prop.design_thrust
+    Power        = prop.design_power
+    
+    airfoil_components = prop.Airfoil_Components
+    airfoil_analyses   = prop.Airfoil_Analyses 
+    a_loc              = prop.airfoil_polar_stations
+    
+    # Check for static conditions
+    if V == 0.0:
+        V = 1E-6 
+    
+    # ---------------------------------------------------------------------------------------------------------
+    # Setup Analyses for Rotor Design
+    # ---------------------------------------------------------------------------------------------------------
+    ## Calculate airfoil properties from airfoil analysis
+    if not bool(airfoil_analyses):
+        raise Exception("Airfoil not yet initialized!")
+        #airfoil_analysis = Airfoil.Airfoil()
+    
+        ## Set prop as geometry for analysis
+        #airfoil_analyses.geometry = prop
+        #airfoil_analyses.initialize()
+        #airfoil_analyses.evaluate()
+    
+    # Calculate atmospheric properties
+    atmosphere = SUAVE.Analyses.Atmospheric.US_Standard_1976()
```

Comment:
remove SUAVE, this is RCAIDE, should be an RCAIDE import

Reply:
fixed